### PR TITLE
docs/dev/libvirt: update libvirt/firewalld setup instructions

### PR DESCRIPTION
**Pull request by travier**
* Update libvirt setup instructions to clearly separate the systemd activated version from the classic one.
  * Update firewalld setup to take into account recent libvirt integration.
  * Also include some minor Markdown styling fixes.

### CODE REVIEW

File: docs/dev/libvirt/README.md [hunk 1]
```diff
@@ -16,13 +17,14 @@ Before you begin, install the [build dependencies](../dependencies.md).
 Make sure you have KVM enabled by checking for the device:
 
 ```console
-$ ls -l /dev/kvm 
+$ ls -l /dev/kvm
```
Comment (cfergeau) [inline docs/dev/libvirt/README.md:20]: Having the whitespace/formatting changes in their own commit would have been nice :)

File: docs/dev/libvirt/README.md [hunk 2]
```diff
@@ -152,34 +163,43 @@ iptables -I INPUT -p tcp -s 192.168.126.0/24 -d 192.168.122.1 --dport 16509 -j A
...
 ```
 
+For system with an older libvirt, we will move the new bridge interface to a
+dedicated network zone and enable incoming libvirt, DNS & DHCP traffic:
+
+```sh
+sudo firewall-cmd --zone=dmz --change-interface=tt0
+sudo firewall-cmd --zone=dmz --add-service=libvirt
+sudo firewall-cmd --zone=dmz --add-service=dns
+sudo firewall-cmd --zone=dmz --add-service=dhcp
```
Comment (cfergeau) [inline docs/dev/libvirt/README.md:192]: Why the change from the previous rules?
```
sudo firewall-cmd --zone=dmz --change-interface=virbr0
sudo firewall-cmd --zone=dmz --change-interface=tt0
sudo firewall-cmd --zone=dmz --add-service=libvirt
```
I assume the rules in this PR are enough to allow connections from the cluster VMs to qemu+tcp://192.168.122.1/system?

File: docs/dev/libvirt/README.md (hunk 1 above)
Comment (travier) [reply docs/dev/libvirt/README.md:20]: Done

File: docs/dev/libvirt/README.md [hunk 3]
```diff
@@ -152,34 +163,43 @@ iptables -I INPUT -p tcp -s 192.168.126.0/24 -d 192.168.122.1 --dport 16509 -j A
 
 #### Firewalld
 
-If using `firewalld`, the specifics will depend on how your distribution setup the
-various zones.
+If using `firewalld`, the specifics will depend on how your distribution setup
```
[the remark this answers is not in the document]
Comment (travier) [reply docs/dev/libvirt/README.md]: Done

File: docs/dev/libvirt/README.md [hunk 4]
```diff
@@ -152,34 +163,43 @@ iptables -I INPUT -p tcp -s 192.168.126.0/24 -d 192.168.122.1 --dport 16509 -j A
...
+the various zones. The following instructions should work as is for Fedora,
+CentOS, RHEL and Arch Linux.
```
[the remark this answers is not in the document]
Comment (travier) [reply docs/dev/libvirt/README.md:168]: I think the "libvirt version less than 5.1" covers EL7. Maybe the phrasing here is misleading?

File: docs/dev/libvirt/README.md (hunk 2 above)
Comment (travier) [reply docs/dev/libvirt/README.md:192]: I have not re-checked this but by default the dmz zone does not allow DNS/DHCP access and the nodes in the cluster need to get DHCP & DNS from the host dnsmasq daemon started by libvirt.

Comment (travier) [reply docs/dev/libvirt/README.md:192]: The `virbr0` change is never required as far as I understand as nodes traffic will get routed through the tt0 bridge.

File: docs/dev/libvirt/README.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (cfergeau) [reply docs/dev/libvirt/README.md:168]: Ah yes, you differentiate between the misc distro versions through libvirt version, which makes sense.

File: docs/dev/libvirt/README.md (hunk 2 above)
Comment (cfergeau) [reply docs/dev/libvirt/README.md:192]: > I have not re-checked this but by default the dmz zone does not allow DNS/DHCP access and the nodes in the cluster need to get DHCP & DNS from the host dnsmasq daemon started by libvirt.

Yep, makes sense, dunno why I did not need that when I tried this :)

> The virbr0 change is never required as far as I understand as nodes traffic will get routed through the tt0 bridge.

I ran some tests, and seems to be fine indeed. Not quite sure why the VMs on the tt0 bridge can reach virbr0 (we need to be able to reach qemu+tcp://192.168.122.1/system from the guest OS).

**Comment by travier** [comment]
Updated to take into account the changes from: https://github.com/openshift/installer/pull/3638.

**Comment by travier** [comment]
Updated to not make the TCP socket unit enabled at boot by default to reduce security issue (i.e. a reboot will default to a safe state on modern systems). I removed the draft state.

File: docs/dev/libvirt/README.md [hunk 5]
```diff
@@ -93,65 +93,71 @@ the libvirt instance on the host machine to deploy workers.
...
+`/etc/default/libvirtd`:
 
 ```
 libvirtd_opts="--listen"
 ```
 
-Next, restart libvirt: `systemctl restart libvirtd`
+Then to enable TCP access to libvirtd, modify `/etc/libvirt/libvirtd.conf` and
+set the following:
+
```
[the remark this answers is not in the document]
Comment (cfergeau) [reply docs/dev/libvirt/README.md:140]: I think I would keep most of the previous order and not move most of the text. Splitting the `libvirt-tcp.socket` discussion from its paragraph could be useful indeed, and being more explicit about what `enable` does is good too.

[the remark this answers is not in the document]
Comment (cfergeau) [reply docs/dev/libvirt/README.md:140]: Ah ok, most of the `libvirtd.conf` changes are not needed with tcp socket activation.`auth_tcp = none` may be needed though?

**Comment by travier** [comment]
Updated to take into account comments. As I reordered some sections, I find it more practical to read the final version than to make sense of the diff: https://github.com/travier/installer/blob/master-docs-dev-libvirt-update/docs/dev/libvirt/README.md

File: docs/dev/libvirt/README.md [hunk 6]
```diff
@@ -91,67 +91,89 @@ components drive deployment of worker machines.  The libvirt cluster-api
...
+```
+listen_tls = 0
+listen_tcp = 1
+auth_tcp = "none"
+tcp_port = "16509"
+```
```
Comment (cfergeau) [inline docs/dev/libvirt/README.md:162]: I'm still a bit dubious with the splitting of the libvirtd.conf instructions, even if they are ignored, it's not harmful to set them in socket-activation setups

File: docs/dev/libvirt/README.md [hunk 7]
```diff
@@ -91,67 +91,89 @@ components drive deployment of worker machines.  The libvirt cluster-api
...
+sudo systemctl restart libvirtd
+```
+
+#### For permanently running libvirt daemon
 
-**NOTE:** if the installation of libvirt included support for socket
-activation via the `libvirt-tcp.socket` systemd unit, the `--listen`
-argument should not be added and thus this step can be skipped.
+This applies only if the libvirt daemon is configured to start and listen on
+UNIX/TCP sockets by itself and not through socket activation.
```
Comment (cfergeau) [inline docs/dev/libvirt/README.md:138]: This reads to me as if this section is when starting libvirtd 'by hand', while it really is about systems using `libvirtd.service` without socket activation (ie without using `libvirtd.socket`)

File: docs/dev/libvirt/README.md (hunk 6 above)
Comment (travier) [reply docs/dev/libvirt/README.md:162]: It's not harmful but it's also unnecessary and might trigger questions. I think we need to have two clearly separated cases to avoid confusion and keeping each case to the point helps with that.

File: docs/dev/libvirt/README.md (hunk 7 above)
Comment (travier) [reply docs/dev/libvirt/README.md]: Updated
